src/Propel/Runtime/Map/RelationMap.php

```diff
@@ -113,10 +113,14 @@ class RelationMap
 
     /**
      * @param string $name Name of the relation.
+     * @param \Propel\Runtime\Map\TableMap $localTable Local table map.
+     * @param \Propel\Runtime\Map\TableMap $foreignTable Foreign table map.
      */
-    public function __construct(string $name)
+    public function __construct(string $name, TableMap $localTable, TableMap $foreignTable)
```

Comment:
Regarding `RelationMap`. I think it is logical that a relation must have two objects between which relationship is. Otherwise, relation is senseless. Now I think It would be better to add `type` to the constructor as well since it is part of relation definition.

Regarding `ColumnMap`. I analyzed a single place with the creation of `ColumpMap` and `ColumpMap` always has newly-added properties. So I think their proper place is in the constructor.

Reply:
But could we accept null as well for BC reasons?
If not assigned it would have to be done via setters for now

This is deprecated (we can soft hint this), but this would break maybe less code?
Or is this always just used within Propel and never via extension?